File: Apriori (Grocery shop).R
#Apriori

# data preprocessing

install.packages('arules')
library(arules)
# Importing the dataset
dataset = read.csv('Market_Basket_Optimisation.csv', header = F)
dataset = read.transactions('Market_Basket_Optimisation.csv', sep = ',', rm.duplicates = T)
summary(dataset)
itemFrequencyPlot(dataset, topN = 10)

# Training Apriori on the dataset

rules = apriori(data = dataset, parameter = list(support = 0.003, confidence = 0.4 ))
# FOR SUPPORT WE WILL CONSIDER PRODUCTS WHICH ARE PURCHASED MINIMUM THREE TIMES A DAY.
# SO IN A WEEK ITS 3*7= 21. SO MINIMUM SUPPPORT : 21/7500 = 0.0028

# VISUALIZATION OF RESULT
inspect(rules[1:10])

#better way is to sort by lift(decrrasing lift)
inspect(sort(rules, by = 'lift')[1:10])

# Here we can see that we are  getting the productes that have high support. the high purchased product. Which is effecting the lift.
#So to get a better combination we will change the confidence.

rules = apriori(data = dataset, parameter = list(support = 0.003, confidence = 0.2 ))

#better way is to sort by lift(decrrasing lift)
inspect(sort(rules, by = 'lift')[1:10])

rules = apriori(data = dataset, parameter = list(support = 0.004, confidence = 0.2 ))
# FOR SUPPORT WE WILL CONSIDER PRODUCTS WHICH ARE PURCHASED MINIMUM FOur TIMES A DAY.
# SO IN A WEEK ITS 3*7= 21. SO MINIMUM SUPPPORT : 28/7500 = 0.0037

#better way is to sort by lift(decrrasing lift)
inspect(sort(rules, by = 'lift')[1:10])

File: Apriori (Grocery shop).py
# -*- coding: utf-8 -*-
"""
Created on Sat Jul 13 16:54:59 2019

@author: SAYELI
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing dataset



# Now we need to prepaire the into correctly. where all the transactions should be a list.
# And all the items in the trasaction should be a list. for that we run a for loop.
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header = None)
transactions = []
for i in range(0, 7501):
    transactions.append([str(dataset.values[i,j]) for j in range(0, 20)])

#Training of the Apriors of the dataset.
from apyori import apriori   
rules = apriori(transactions, min_support =0.003, min_confidence = 0.20, min_lift = 3, min_lenght = 2 )
''' We Considered a product that is purchased three times a day, So for a week its
3*7 = 21. We have total 7501 transaction. So min_support = 21/7501 = 0.0027'''

''' We have seen earlier that minimum confidence when set high takes products which
are bought very frequently not because they are truly associated instead they 
purchased more number of time'''

#Visualization of the results

results = list(rules)

output = []
for row in results:
    output.append(str(row.items))